modules/loader-utils/src/types.ts

```diff
@@ -204,7 +204,7 @@ export type LoaderContext = {
     loaders?,
     options?: LoaderOptions,
     context?: LoaderContext
-  ) => AsyncIterable<any>;
+  ) => Promise<AsyncIterable<any>>;
```

Comment:
`parseInBatches` returns this type:
https://github.com/visgl/loaders.gl/blob/923e4969ee3110d7d6c0c4bcd4e7c44daf054feb/modules/core/src/lib/api/parse-in-batches.ts#L32

Reply:
At the same time custom parsers return just `AsyncIterable`:
https://github.com/visgl/loaders.gl/blob/923e4969ee3110d7d6c0c4bcd4e7c44daf054feb/modules/shapefile/src/lib/parsers/parse-dbf.ts#L126

Reply:
Thanks for tracking that down. This is what I was "afraid of": the fact that there are no type errors here should be fixed first. Otherwise any change we make to defeat warnings is based on weak foundations. 

A next question would be: why do we not get a warning here:

https://github.com/visgl/loaders.gl/blob/923e4969ee3110d7d6c0c4bcd4e7c44daf054feb/modules/shapefile/src/dbf-loader.ts#L32

Note that I am now avoiding declaring the type of the loader directly, as that removes information from the exported type.

Instead I check types like this https://github.com/visgl/loaders.gl/blob/923e4969ee3110d7d6c0c4bcd4e7c44daf054feb/modules/shapefile/src/shapefile-loader.ts#L32

Maybe that makes a difference?